jira/resilientsession.py

```diff
@@ -123,103 +266,31 @@ def __recoverable(
-            exception = None
-            try:
-                method = getattr(super(), verb.lower())
-                response = method(url, timeout=self.timeout, **kwargs)
-                if response.status_code >= 200 and response.status_code <= 299:
-                    return response
-            except ConnectionError as e:
-                logging.warning(f"{e} while doing {verb.upper()} {url}")
-
-                exception = e
-            retry_number += 1
-
-            if retry_number <= self.max_retries:
-                response_or_exception = response if response is not None else exception
-                if self.__recoverable(
-                    response_or_exception, url, verb.upper(), retry_number
-                ):
-                    if retry_data:
-                        # if data is a stream, we cannot just read again from it,
-                        # retry_data() will give us a new stream with the data
-                        kwargs["data"] = retry_data()
-                    continue
-                else:
-                    break
 
-        if exception is not None:
-            raise exception
-        raise_on_error(response, verb=verb, **kwargs)
-        # after raise_on_error, only Response objects are allowed through
-        response = cast(Response, response)  # tell mypy only Response-like are here
-        return response
-
-    def get(self, url: Union[str, bytes], **kwargs) -> Response:  # type: ignore
-        return self.__verb("GET", str(url), **kwargs)
-
-    def post(self, url: Union[str, bytes], data=None, json=None, **kwargs) -> Response:  # type: ignore
-        return self.__verb("POST", str(url), data=data, json=json, **kwargs)
-
-    def put(self, url: Union[str, bytes], data=None, **kwargs) -> Response:  # type: ignore
-        return self.__verb("PUT", str(url), data=data, **kwargs)
-
-    def delete(self, url: Union[str, bytes], **kwargs) -> Response:  # type: ignore
-        return self.__verb("DELETE", str(url), **kwargs)
-
-    def head(self, url: Union[str, bytes], **kwargs) -> Response:  # type: ignore
-        return self.__verb("HEAD", str(url), **kwargs)
-
-    def patch(self, url: Union[str, bytes], data=None, **kwargs) -> Response:  # type: ignore
-        return self.__verb("PATCH", str(url), data=data, **kwargs)
+        if is_recoverable:
+            # Exponential backoff with full jitter.
+            delay = min(self.max_retry_delay, 10 * 2**counter) * random.random()
+            LOG.warning(
+                "Got recoverable error from %s %s, will retry [%s/%s] in %ss. Err: %s"
+                % (request_method, url, counter, self.max_retries, delay, msg)  # type: ignore[str-bytes-safe]
+            )
+            if isinstance(response, Response):
+                LOG.debug(
+                    "Dumping Response headers and body. "
+                    + f"Log level debug in '{__name__}' is not safe for production code!"
+                )
```

Comment:
I wonder if this should be in the `__init__` as maybe if it gets here it is too late to warn the developer about it?